### PR TITLE
fix(k8s): hash configuration annotation

### LINKED ISSUE #1496: garden plugins kubernetes cleanup-cluster-registry fails with metadata.annotations: Too long: must have at most 262144 characters
### Current Behavior

`garden plugins kubernetes cleanup-cluster-registry `fails with:
`The Deployment "garden-docker-registry" is invalid: metadata.annotations: Too long: must have at most 262144 characters`

`garden.io/last-applied-configuration` is the thing that gets unreasonably long.

### Expected behavior

It should not fail and restart the registry successfully.

### Reproducible example

Run `garden plugins kubernetes cleanup-cluster-registry` multiple times. 

### Workaround

Delete garden.io/last-applied-configuration in the deployment configuration on the cluster.

### Your environment

`garden version`: 0.11.0-alpha.4

### PR DESCRIPTION
**Pull request by eysi09**
**What this PR does / why we need it**:

Previously we stored the raw manifest under the
garden.io/last-applied-configuration. We now hash it and store it under
garden.io/manifest-hash. This prevents errors when then annotation
string gets too long.

**Which issue(s) this PR fixes**:

Fixes #1496

### CODE REVIEW

File: garden-service/src/plugins/kubernetes/util.ts [hunk 1]
```diff
@@ -19,6 +19,8 @@ import { ContainerEnvVars } from "../container/config"
 import { ConfigurationError } from "../../exceptions"
 import { KubernetesProvider } from "./config"
 import { LogEntry } from "../../logger/log-entry"
+import hasha from "hasha"
+import { stringify } from "querystring"
```
Comment (edvald) [inline garden-service/src/plugins/kubernetes/util.ts]: I think that's the wrong import. Previous iteration used a stable JSON serialization.

Comment (eysi09) [reply garden-service/src/plugins/kubernetes/util.ts]: Good catch! Just used the auto-import feature and obviously wasn't paying attention.

File: garden-service/src/plugins/kubernetes/util.ts [hunk 2]
```diff
@@ -19,6 +19,8 @@ import { ContainerEnvVars } from "../container/config"
 import { ConfigurationError } from "../../exceptions"
 import { KubernetesProvider } from "./config"
 import { LogEntry } from "../../logger/log-entry"
+import hasha from "hasha"
+import stringify from "json-stringify-safe"
```
Comment (edvald) [inline garden-service/src/plugins/kubernetes/util.ts]: Still the wrong library. `json-stable-stringify` is the one, to ensure that it's deterministic.

Comment (eysi09) [reply garden-service/src/plugins/kubernetes/util.ts]: 🤦‍♂️

**Pull request merged**